### ISSUE TITLE
Discount is lost when going back to item list

### ISSUE THREAD

**Issue by rdantonio** [paypal/here-sideloader-api-samples#29]
I tested this on Android, but have not had a chance to test on iOS yet.

I have an invoice that I am passing discountAmount on the invoice level (I tested and this also happens with discountPercent).

The discount is properly applied to the invoice initially, but if I click Charge to open the payment method chooser and then dismiss the chooser without selecting an option by using the "X", the discount is lost and it displays the un-discounted full total.

This also happens if you select a payment method (ex: cash), hit the back arrow on the cash received page, then hit the "X" on the chooser.

Also, from what I can see the discount is not displayed anywhere. Not sure if this is intended, but it would be nice to have unless I missed it somewhere.

Thanks!

**Comment by lernerb** [comment]
While logged in on Android, can you go to your settings page and let me know if the discounts switch is turned to `on` or `off`?

I'll add this to our backlog & investigate.

**Comment by rdantonio** [comment]
Oops, that did the trick. It was off. Would it be possible to turn it on automatically if discountAmount or discountPercent is passed through the API?

If not I think the discount shouldn't be applied initially then, to avoid confusion?

**Comment by lernerb** [comment]
Thought that would do it :) I'll bring this up with our product team.

**Comment by rdantonio** [comment]
Awesome, thank you!

**Comment by rdantonio** [comment]
I ran a quick test on iOS. In the settings, I left Discount set to "off".

When I posted my transaction to HERE, it displayed the Discount line as if the Discount setting was set to "on" and everything behaved as expected (the discount did not get cleared when "X"ing out the payment method selector). Once the transaction was complete, further transactions in HERE independent of sideloader APIs did not have the Discount line.

It seems like iOS is turning on the Discount functionality just for the transaction if discountAmount or discountPercent is passed through the API.

This would be my preferred behavior on Android as well, if possible, unless this is not intended on iOS :)

**Comment by lernerb** [comment]
Just to confirm, you turned the `discount` switch to `off` before you ran this test?

**Comment by rdantonio** [comment]
Correct. I turned the Discount switch to off, killed the app, relaunched, verified that it was still off, then ran the test (just to be absolutely 100% sure).
